### PR TITLE
Added a "display_on_attach" option.

### PR DESCRIPTION
**Pull request by dc-m**
This allows you to set true or false. If true it will show all available unfiltered items. If false it will not show any item until you start typing (like original).

### CODE REVIEW

**Comment by hakib** [comment]
Hey @dc-m, Thanks for the PR!

The diff is very hard to read (is it possible you formated the entire file?).

I left some comments but I might have missed some changes as all the file is marked as changed. If you could push an update that will highlight only the changes you made it will be much easier for me to review.

**Comment by dc-m** [comment]
Sure I will do that when i get back from work tonight - sorry i CTRL + K + D and it must have caused a whole file reshift!

File: massautocomplete.js [hunk 1]
```diff
@@ -1,344 +1,348 @@
...
+        current_element = target_element;
+        current_model = ngmodel;
+        current_options = options;
+        previous_value = ngmodel.$viewValue;
+
+        $scope.results = [];
+        $scope.selected_index = -1;
+        bind_element();
+
+        //Added to ensure on attach if value there it doesnt try reset to all - DAVID 08/06/2017
```
Comment (dc-m) [inline massautocomplete.js]: Change

File: massautocomplete.js [hunk 2]
```diff
@@ -1,344 +1,348 @@
...
+        if (current_element[0].value.length == 0) suggest("", current_element);;
+
+        value_watch = $scope.$watch(
+          function () {
+            if (current_options.display_on_attach && ngmodel.$modelValue == null) ngmodel.$modelValue = "";
```
Comment (dc-m) [inline massautocomplete.js]: Change

File: massautocomplete.js [hunk 3]
```diff
@@ -1,344 +1,348 @@
...
+          }
+        );
+      }
+      that.attach = debounce(_attach, user_options.debounce_attach);
+
+      function _suggest(term, target_element) {
+        $scope.selected_index = 0;
+        $scope.waiting_for_suggestion = true;
+
+        if (typeof(term) === 'string' && term.length > 0 || current_options.display_on_attach) {
```
Comment (dc-m) [inline massautocomplete.js]: change

**Comment by dc-m** [comment]
Apologies for the way I've done this.

I attempted to get it to only show my changes but each time I did it seems to just remove all the original and add all my file. So have commented where I have made changes

File: massautocomplete.js [hunk 4]
```diff
@@ -1,344 +1,348 @@
...
+        current_model = ngmodel;
+        current_options = options;
+        previous_value = ngmodel.$viewValue;
+
+        $scope.results = [];
+        $scope.selected_index = -1;
+        bind_element();
+
+        //Added to ensure on attach if value there it doesnt try reset to all - DAVID 08/06/2017
+        if (current_element[0].value.length == 0) suggest("", current_element);;
```
Comment (dc-m) [inline massautocomplete.js]: Change

**Comment by hakib** [comment]
Hey David, 

A more general approach would be "suggest_when_empty" option. This way you we can also address the situation where the user deleted all the text from the input and you want to trigger a suggest cycle to show some options.

If you set `last_selected_value` to null in `on_attach` for example, it will trigger a suggest cycle when you first focus the field (given that there is no value to the ngModel).

```
        if (options.suggest_when_empty) {
          // Set last value to something different than th initial value
          // so the first suggest will trigger suggest.
          last_selected_value = null;
        }
```

**Pull request closed without merging**